internal/deployers/eksapi/infra.go

```diff
@@ -406,11 +414,12 @@ func (m *InfrastructureManager) getAZsWithInstanceTypes(opts *deployerOptions) (
 		counts[az] = count + 1
 	}
 	var azs []string
-	for az, count := range counts {
-		if count == len(opts.InstanceTypes) {
-			azs = append(azs, az)
-		}
+	for az := range counts {
+		azs = append(azs, az)
 	}
+	sort.Slice(azs, func(i, j int) bool {
+		return counts[azs[i]] > counts[azs[j]]
+	})
```

Comment:
I think this makes sense, but I think we should also explicitly filter out AZs that do not support any of the specified instance types, it'll just make the logic more reliable if/when we change `numAZs`.